src/telemetryReporter.ts

```diff
@@ -18,9 +20,16 @@ export default class TelemetryReporter extends vscode.Disposable {
     private static TELEMETRY_CONFIG_ID = 'telemetry';
     private static TELEMETRY_CONFIG_ENABLED_ID = 'enableTelemetry';
 
+    private logFilePath: string;
+    private logStream: fs.WriteStream;
+
     constructor(private extensionId: string, private extensionVersion: string, key: string) {
         super(() => this.toDispose.forEach((d) => d && d.dispose()))
-
+        this.logFilePath = process.env['VSCODE_LOGS'] || '';
+        if (this.logFilePath) {
+            this.logFilePath = path.join(this.logFilePath, `${extensionId}.txt`);
+            this.logStream = fs.createWriteStream(this.logFilePath, { flags: 'a', encoding: 'utf8' });
```

Comment:
We should close the stream on dispose.

Reply:
Done. I've used `end()` which will flush out all the pending writes. Then waited for it to "finish" before returning the promise in `dispose`